protocol/x/prices/keeper/msg_server_create_oracle_market.go

```diff
@@ -2,14 +2,37 @@ package keeper
 
 import (
 	"context"
-	"errors"
 
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 	"github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
 )
 
 func (k msgServer) CreateOracleMarket(
 	goCtx context.Context,
 	msg *types.MsgCreateOracleMarket,
 ) (*types.MsgCreateOracleMarketResponse, error) {
-	return &types.MsgCreateOracleMarketResponse{}, errors.New("Not implemented")
+	if !k.Keeper.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Use zero oracle price to create the new market.
+	// Note that valid oracle price updates cannot be zero (checked in MsgUpdateMarketPrices.ValidateBasic),
+	// so a zero oracle price indicates that the oracle price has never been updated.
+	zeroMarketPrice := types.MarketPrice{
```

Comment:
Do we have any tests that explicitly confirm that zero prices are ignored by clob, etc as unusable? Esp. integration / e2e.

Related question: do you have a test plan for adding a new market and seeing trading delayed until a price update occurs?

Reply:
Yes, we will cover e2e test in this [project](https://linear.app/dydx/project/[m5c-p0]-core-user-flow-test-coverage-5e4f1dc1cfef)! Will document test plan as part of this [ticket](https://linear.app/dydx/issue/CORE-571/e2e-test-for-gov-add-new-markets)